src/main/scala/mesosphere/util/state/PersistentStore.scala

```diff
@@ -0,0 +1,80 @@
+package mesosphere.util.state
+
+import scala.concurrent.Future
+
+/**
+  * An Entity that is provided and can be changed by the related store.
+  */
+trait PersistentEntity {
+
+  /**
+    * The identifier of this entity.
+    */
+  def id: String
+
+  /**
+    * Get the content bytes of this entity.
+    */
+  def bytes: Array[Byte]
+
+  /**
+    * Mutate this entity with given bytes.
+    */
+  def mutate(bytes: Array[Byte]): PersistentEntity
```

Comment:
different name and description: This method does not actually modify the instance itself but returns a new instance.